### PR TITLE
Remove Numpy as a requirement

### PR DESCRIPTION
**Pull request by mundya**
I'm not sure how we want to test this - we could create a new Tox environment that doesn't install Numpy and check that the method isn't available...?

### CODE REVIEW

File: rig/type_casts.py [hunk 1]
```diff
@@ -264,3 +178,97 @@ def validate_fp_params(signed, n_bits, n_frac):
     max_v = ((1 << n_int) - 1) / float(1 << n_frac)
 
     return min_v, max_v
+
+
+if np is not None:
```
Comment (mossblaser) [inline rig/type_casts.py]: # pragma: no branch

File: rig/type_casts.py [hunk 2]
```diff
@@ -1,6 +1,9 @@
 """Fixed point conversion utilities.
 """
-import numpy as np
+try:
+    import numpy as np
+except ImportError:
+    np = None  # No Numpy support
```
Comment (mossblaser) [inline rig/type_casts.py]: # pragma: no cover

File: setup.py [hunk 3]
```diff
@@ -12,8 +12,9 @@
     license="GPLv2",
 
     # Requirements
-    install_requires=["numpy>1.6", "six", "enum34", "sentinel"],
+    install_requires=["six", "enum34", "sentinel"],
     tests_require=["pytest>=2.6", "pytest-cov", "mock", "toposort"],
```
Comment (mossblaser) [inline setup.py]: tests need "numpy>1.6"

**Comment by mossblaser** [comment]
An alternative for could be to monkeypatch numpy out in the tests file and re-import the module and verify the functions are missing in a test. That way you don't need a whole test run to check the behaviour.

Incidentally, might it not be better from the principle of least surprise point of view to replace numpy-dependencies with stubs which produce a suitably informative exception? (Could be done with a decorator too, and everyone loves decorators!)

**Comment by mundya** [comment]
Decorator sounds good, I'll get on that.

**Comment by mossblaser** [comment]
Mutually agreed that this idea will be shelved for now due to pains in testing etc.

**Pull request closed without merging**